modules/help_editor/jsx/help_editor.js

```diff
@@ -0,0 +1,168 @@
+
+/**
+ * Help Editor Archive Page.
+ *
+ * Serves as an entry-point to the module, rendering the whole react
+ * component page on load.
+ *
+ * Renders Help Editor main page consisting of FilterTable and
+ * DataTable components.
+ *
+ * @author LORIS Team
+ * @version 1.0.0
+ *
+ * */
+class HelpEditor extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoaded: false,
+      filter: {},
+      data: {}
+    };
+
+    /*
+     * Bind component instance to custom methods
+     */
+    this.fetchData    = this.fetchData.bind(this);
+    this.updateFilter = this.updateFilter.bind(this);
+    this.resetFilters = this.resetFilters.bind(this);
+    this.formatColumn = this.formatColumn.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   */
+  fetchData() {
+    $.ajax(this.props.DataURL, {
+      method: 'GET',
+      dataType: 'json',
+      success: data => {
+	      this.setState({
+          data: data,
+          isLoaded: true
+        });
+      },
+      error: error => console.error(error)
+    });
+  }
+
+  /**
+   * Set this.state.filter to the input filter object
+   *
+   * @param {object} filter - the filter object
+   */
+  updateFilter(filter) {
+    this.setState({filter});
+  }
+
+  //TODO: Clearing filters via refs should be deprecated in future refactoring.
+  /**
+   * Reset the filter elements with textInput refs to empty values
+   */
+  resetFilters() {
+    this.filter.clearFilter();
+  }
+
+  /**
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {arrray} rowData - array of cell contents for a specific row
+   * @param {arrray} rowHeaders - array of table headers (column names)
+   *
+   * @return {*} a formated table cell for a given column
+   */
+  formatColumn(column, cell, rowData, rowHeaders) {
+    if (this.state.data.hiddenHeaders.indexOf(column) > -1) {
+      return null;
+    }
+
+    // Create the mapping between rowHeaders and rowData in a row object.
+    let row = {};
+    let url;
+    rowHeaders.forEach((header, index) => row[header] = rowData[index]);
+  
+    if (column === 'Topic') {
+      url = loris.BaseURL + '/help_editor/edit_help_content/?helpID=' +
+             row.HelpID + '&parentID=' + row.ParentID;
+      return <td><a href ={url}>{cell}</a></td>;
+    }
+    if (column === 'Parent Topic') {
+      url = loris.BaseURL + '/help_editor/edit_help_content/?helpID=' +
+             row.ParentID + '&parentID=' + row.ParentTopicID;
+      return <td><a href ={url}>{cell}</a></td>;
+    }
+  
+    return <td>{cell}</td>;
+  }
+
+  render() {
+    // Waiting for async data to load
+    if (!this.state.isLoaded) {
+      return <Loader/>
+    }
+
+    //TODO: refs should be deprecated in future refactoring.
```

Comment:
is this still relevant?

Reply:
my understanding is that string refs are being deprecated, but callback refs are fine, which is what you're introducing here. so the todo seems to be resolved here to me. @davidblader what do you think?

Reply:
you are both correct. removing the clearFilter refs is a longterm goal since they are not really doing anything that would require a ref, but in the mean time the callback pattern should be used since the string format is prone to be dropped for any upcoming React release. 
i think the comment can be left as is